src/riak_core_info_service_process.erl

```diff
@@ -0,0 +1,113 @@
+-module(riak_core_info_service_process).
```

Comment:
At this point you've pretty much rolled your own `gen_server`, so it doesn't seem to me that the argument against using the OTP behavior has any basis left.